code/modules/martial_arts/combos/adv_first_aid/mend_bleeding.dm

```diff
@@ -0,0 +1,21 @@
+/datum/martial_combo/adv_first_aid/mend_bleeding
+	name = "Mend Artery"
+	steps = list(MARTIAL_COMBO_STEP_HELP, MARTIAL_COMBO_STEP_DISARM, MARTIAL_COMBO_STEP_GRAB, MARTIAL_COMBO_STEP_HARM)
+	explaination_text = "Mends an instance of internal bleeding in the targeted limb."
+
+/datum/martial_combo/adv_first_aid/mend_bleeding/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	if(!do_after(user, 10 SECONDS, TRUE, target))
+		to_chat(usr, "<span class='warning' you were interrupted. </span>")
+		return
+
+	if(!target.stat && ishuman(target))
+		target.visible_message("<span class='warning'>[user] creates a tourniquet using [target]'s clothing. </span>", \
+								"<span class='userdanger'>[user] makes a tourniquet out of your clothes.</span>")
+		var/obj/item/organ/external/affected = target.get_organ(user.zone_selected)
+		if(affected.internal_bleeding)
+			target.apply_damage(10, BRUTE)
+			target.Slowed(5)
+			affected.internal_bleeding = FALSE
+		add_attack_logs(user, target, "Healed with martial-art [MA] : [src]", ATKLOG_ALL)
```

Comment:
```suggestion
		var/obj/item/organ/external/affected = target.get_organ(user.zone_selected)
		if(affected && affected.internal_bleeding)
			target.visible_message("<span class='warning'>[user] creates a tourniquet using [target]'s clothing, staunching the bleeding in [target.p_their()] [affected.name].</span>", \
							   	   "<span class='userdanger'>[user] makes a tourniquet out of your clothes, staunching the bleeding in your [affected.name].</span>")
			target.apply_damage(10, BRUTE)
			target.Slowed(5)
			affected.internal_bleeding = FALSE
			add_attack_logs(user, target, "Healed with martial-art [MA] : [src]", ATKLOG_ALL)
```

The limb might've disappeared during the progress bar, so you need to check if it's still there. 

Additionally edited the message to indicate which limb got affected.

Reply:
Should be fixed now, @ me again if I still missed something, it's a little hard to read the suggestion due to formatting on github sorry about the hassle.